### PR TITLE
Fix/ios wrong permission message

### PR DESCRIPTION
**Pull request by ishpaul777**
### Details

Fixed the incorrect error message displaying "camera permission denied" when the actual issue was a denial of microphone permission. Now, the error message accurately reflects the microphone permission denial.

### Fixed Issues

$ https://github.com/Expensify/App/issues/23421
PROPOSAL: https://github.com/Expensify/App/issues/23421#issuecomment-1650414468

### Tests

- [x] Verify that no errors appear in the JS console
Can be tested on ios
- [x] Complete 'Add Bank Account" flow until step 3. 
- [x] Decline Camera permission in Settings. Start Onfido Verfication, Verfify error with "camera permission denied" (exact copy needs approval [here](https://github.com/Expensify/App/issues/23421#issuecomment-1779520038)) is shown.
- [x] Decline Microphone permission in Settings. Start Onfido Verfication, Verfify error with "Microphone permission denied" (exact copy needs approva [herel](https://github.com/Expensify/App/issues/23421#issuecomment-1779520038)) is shown.

### QA Steps

- [x] Verify that no errors appear in the JS console
Can be tested on ios and android
- [x] Complete 'Add Bank Account" flow until step 3. 
- [x] Decline Camera permission in Settings. Start Onfido Verfication, Verfify error with "camera permission denied" (exact copy needs approval [here](https://github.com/Expensify/App/issues/23421#issuecomment-1779520038)) is shown.
- [x] Decline Microphone permission in Settings. Start Onfido Verfication, Verfify error with "Microphone permission denied" (exact copy needs approva [herel](https://github.com/Expensify/App/issues/23421#issuecomment-1779520038)) is shown.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/104348397/17c0be12-825a-484a-9eca-869d1245bd10

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/104348397/3a673d4d-64f7-4bb7-b00a-2017a0b165f2

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/Onfido/index.native.js [hunk 1]
```diff
@@ -39,30 +40,53 @@ function Onfido({sdkToken, onUserExit, onSuccess, onError}) {
...
-                                    Linking.openSettings();
-                                },
-                            },
-                        ],
-                        {cancelable: false},
-                    );
-                    return;
-                }
+                if (!_.isEmpty(errorMessage)) {
+                    const micPermission = Platform.select({
```
Comment (situchan) [inline src/components/Onfido/index.native.js:44]: `Platform.select` is not preferred. Please re-use `iou/ReceiptSelector/CameraPermission` structure to handle platform specific permissions.

**Comment by situchan** [comment]
@ishpaul777 you won't be able to test adhoc build as your udid isn't registered.
I will test myself. I think C+ testing should be enough.

**Comment by situchan** [comment]
> @situchan can you please explain how can i build app on ios physical device it would be helpful for future issue, do i need some extra permissions or what?

If you have your own apple developer account or got invited from any other team, you can point to that cert/provisioning profile, change bundle id to any random one and install the app on iPhone. Note: it's not possible to test features which require cert, i.e. push notification.
If you don't have access to any development team, it's not possible.

File: src/components/Onfido/index.native.js [hunk 2]
```diff
@@ -39,30 +41,52 @@ function Onfido({sdkToken, onUserExit, onSuccess, onError}) {
...
-                                    onUserExit();
-                                    Linking.openSettings();
-                                },
-                            },
-                        ],
-                        {cancelable: false},
-                    );
-                    return;
-                }
+                const os = getPlatform();
```
Comment (situchan) [inline src/components/Onfido/index.native.js:44]: can we use platform files? index.android.js, index.ios.js

File: src/languages/en.ts [hunk 3]
```diff
@@ -1217,8 +1217,10 @@ export default {
         tryAgain: 'Try again',
         verifyIdentity: 'Verify identity',
         genericError: 'There was an error while processing this step. Please try again.',
-        cameraPermissionsNotGranted: 'Camera permissions not granted',
-        cameraRequestMessage: 'You have not granted us camera access. We need access to complete verification.',
+        cameraPermissionsNotGranted: 'Enable Camera Access',
+        cameraRequestMessage: 'We need access to your camera to complete bank account verification. Please enable via Settings > New Expensify',
+        microphonePermissionsNotGranted: 'Enable Microphone Access',
```
Comment (situchan) [inline src/languages/en.ts:1222]: > - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.

This is already approved but just to double confirm that this is not treated as sentence and capitalization of each word is fine.
cc: @mallenexpensify

**Comment by situchan** [comment]
> > can we use platform files? index.android.js, index.ios.js
> 
> @situchan we check permission for ios only right, i believe the pattern is more useful when we want to use utility for different platforms also this was approved [here](https://github.com/Expensify/App/issues/23421#issuecomment-1789733862) and there are places we already using `getPlatform` for platform spectific conditionals.

👌 

Author Checklist failing. Please pull main

**Comment by situchan** [comment]
Still failing. @ishpaul777 please pull main again

**Comment by chiragsalian** [comment]
Author checklist is failing because,
1. @ishpaul777 has not completed the author checklist [here](https://github.com/Expensify/App/pull/30374#issue-1961950035)
2. It could be out of date and if that's the case refer the latest [here](https://github.com/Expensify/App/blob/32a29b1b2ebd2686ecf9d9ff3d72a649661ad428/.github/PULL_REQUEST_TEMPLATE.md)

**Comment by allroundexperts** [comment]
> @allroundexperts, sorry for ping again. I just wanted to check in and see if you might have an estimated time when you can test with these [cases](https://github.com/Expensify/App/pull/30374#issuecomment-1817462416) its been on hold for a while. Thanks!

@ishpaul777 I don't think that these test cases are possible for me to test. As soon as you go offline, we're blocking the Onfido flow.

File: src/components/Onfido/index.native.js [hunk 4]
```diff
@@ -39,30 +41,52 @@ function Onfido({sdkToken, onUserExit, onSuccess, onError}) {
...
+                if (!_.isEmpty(errorMessage) && os === CONST.PLATFORM.IOS) {
```
Comment (situchan) [inline src/components/Onfido/index.native.js:45]:
```suggestion
                if (!_.isEmpty(errorMessage) && getPlatform() === CONST.PLATFORM.IOS) {
```

File: src/components/Onfido/index.native.js [hunk 5]
```diff
@@ -39,30 +41,52 @@ function Onfido({sdkToken, onUserExit, onSuccess, onError}) {
...
+                            let alertTitle = '';
+                            let alertMessage = '';
+                            if (!isMicAllowed) {
+                                alertTitle = 'onfidoStep.microphonePermissionsNotGranted';
+                                alertMessage = 'onfidoStep.microphoneRequestMessage';
+                            }
+                            if (!isCameraAllowed) {
+                                alertTitle = 'onfidoStep.cameraPermissionsNotGranted';
+                                alertMessage = 'onfidoStep.cameraRequestMessage';
+                            }
```
Comment (situchan) [inline src/components/Onfido/index.native.js:59]:
```suggestion
                            if (!isCameraAllowed) {
                                alertTitle = 'onfidoStep.cameraPermissionsNotGranted';
                                alertMessage = 'onfidoStep.cameraRequestMessage';
                            } else if (!isMicAllowed) {
                                alertTitle = 'onfidoStep.microphonePermissionsNotGranted';
                                alertMessage = 'onfidoStep.microphoneRequestMessage';
                            }
```

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/108292595/5dfcef94-6b55-451b-811a-b31284b9e42e


https://github.com/Expensify/App/assets/108292595/5a289f54-9098-4198-a14f-5990ed828366



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/108292595/0e0c9a52-3563-4513-a9ac-a4a8c21c8c6c


https://github.com/Expensify/App/assets/108292595/f27d01be-6332-4b36-9d79-b42e1a5fe149



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by situchan — APPROVED** [review]
Some NAB suggestions:

https://github.com/Expensify/App/blob/4f41819632e24729f58d2e17340d4549d8d9cca6/src/CONST.ts#L1093-L1097

remove these lines as not used anymore

File: src/languages/en.ts [hunk 6]
```diff
@@ -1014,7 +1014,7 @@ export default {
                 label: 'Use Device Settings',
             },
         },
-        chooseThemeBelowOrSync: 'Choose a theme below, or sync with your device settings.',
+        chooseThemeBelowOrSync: 'Choose a theme below, or syn c with your device settings.',
```
Comment (situchan) [inline src/languages/en.ts:1017]: undo this change
```suggestion
        chooseThemeBelowOrSync: 'Choose a theme below, or sync with your device settings.',
``

File: src/languages/en.ts [hunk 7]
```diff
@@ -1015,6 +1015,7 @@ export default {
             },
         },
         chooseThemeBelowOrSync: 'Choose a theme below, or sync with your device settings.',
+``
```
Comment (situchan) [inline src/languages/en.ts:1018]:
```suggestion
```

**Comment by chiragsalian** [comment]
I'm OOO so i'm unsure if i can review this in a timely manner once it's ready for review. So, I am unassigning myself and once its ready for review please reassign to another internal Expensify engineer for review.

**Comment by chiragsalian** [comment]
oops i just realized this is ready for review. I think it was still WIP, looking at it now 🙂

**Review by chiragsalian — APPROVED** [review]
LGTM
